### PR TITLE
Fixes multiple issues related to compilation problems in new 0.3.0 version.

### PR DESCRIPTION
**Pull request by matux**
## Description of the change

_There's still an open issue reported by_ `pana` _regarding the new_ `rollbar_common` _library which doesn't hasn't been published, yet._

Adds support for Flutter 3.x
Removed unsupported Dart and Flutter versions
Fixes iOS support
Fixes dependency issues
Fixes CI
Adds some extra configuration for FVM
Fixes linter and pana issues.

Preparing `rollbar_common` 0.3.0-beta for publishing so all pana checks pass.
Added MIT `LICENSE` file to `rollbar_common`

Part of 0.3.0-beta update.

## Type of change

- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Maintenance
- [x] New release

## Related issues

> Shortcut stories and GitHub issues (delete irrelevant)

- Fix https://github.com/rollbar/rollbar-flutter/issues/25
- Fix https://github.com/rollbar/rollbar-flutter/issues/24
- Fix https://github.com/rollbar/rollbar-flutter/issues/19

### Development

- [x] Lint rules pass locally
- [x] The code changed/added as part of this pull request has been covered with tests
- [x] All tests related to the changed code pass in development

### Code review

- [x] This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers assigned
- [x] Issue from task tracker has a link to this pull request
- [ ] Changes have been reviewed by at least one other engineer

### CODE REVIEW

**Comment by matux** [comment]
> LGTM, seems like a bit of cleanup and standardization.

Mostly, yes. Only remaining task is to publish `rollbar_common`. All needed to release `0.3.0-beta`.

Can't wait to start chugging PRs with features.

**Comment by matux** [comment]
Successfully published `rollbar-common` on pub.dev, previous checks are now passing, but found new compilation issues I'm dealing with now.

**Review by akornich — COMMENTED** [review]
LGTM with a few minor comments/questions

File: rollbar_flutter/example/ios/Podfile.lock [hunk 1]
```diff
@@ -1,29 +1,39 @@
...
+    :path: ".symlinks/plugins/connectivity_plus/ios"
   Flutter:
     :path: Flutter
   rollbar_flutter:
     :path: ".symlinks/plugins/rollbar_flutter/ios"
 
 SPEC CHECKSUMS:
+  connectivity_plus: 413a8857dd5d9f1c399a39130850d02fe0feaf7e
   Flutter: 50d75fe2f02b26cc09d224853bb45737f8b3214a
+  ReachabilitySwift: 985039c6f7b23a1da463388634119492ff86c825
```
[the remark this answers is not in the document]
Comment (matux) [reply rollbar_flutter/example/ios/Podfile.lock:33]: It seems the new Flutter version makes use of it.

File: rollbar_common/pubspec.yaml [hunk 2]
```diff
@@ -1,22 +1,20 @@
 name: rollbar_common
-description: Package used by the rest of the SDK's packages.
+description: Commons package used by the Rollbar Dart and Flutter SDKs for error reporting.
```
[the remark this answers is not in the document]
Comment (matux) [reply rollbar_common/pubspec.yaml:2]: I'm not sure I understand, what list?

File: rollbar_common/lib/src/_internal/module.dart [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import 'package:logging/logging.dart';
 import 'package:meta/meta.dart';
-import '../sdk_logger.dart';
+import 'package:rollbar_common/src/sdk_logger.dart';
```
[the remark this answers is not in the document]
Comment (matux) [reply rollbar_common/lib/src/_internal/module.dart:3]: Can't find anything on the [Dart style](https://dart.dev/guides/language/effective-dart/style) guidelines, but I agree. I may have done this while playing with the language while learning it, and left it.

**Comment by JayronKhong** [comment]
great package but can we have a 0.3.0 release ?

**Pull request merged**